api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -372,6 +372,41 @@ type NodePoolManagement struct {
 	// +optional
 	// +kubebuilder:default=false
 	AutoRepair bool `json:"autoRepair"`
+
+	// AutoRepairSettings health checks settings for machines in the NodePool when AutoRepair is enabled.
+	//
+	// +optional
+	AutoRepairSettings *AutoRepairSettings `json:"autoRepairSettings,omitempty"`
+}
+
+type AutoRepairSettings struct {
```

Comment:
I think at most we would want to surface UnhealthyConditions and NodeStartupTimeout.
MaxUnhealthy / UnhealthyRange are coming from an initial design were the mhc implemented default remediation itself. Then we moved default remediation into the MachineSet and there's ongoing work upstream for extending remediation capabilities for e.g. limit inflight remediations.

Reply:
I thought since Cesar introduced `MachineHealthCheckMaxUnhealthyAnnotation` annotation to override `MaxUnhealthy` there is a use-case already for users to set a different value. 
Do you mean that MaxUnhealthy / UnhealthyRange will be removed/deprecated upstream soon?

Reply:
@muraee I introduced it as an escape hatch for situations in which there's a lot of unhealthy nodes that should be recycled (this occurred once in rosa production already).

Reply:
removed MaxUnhealthy / UnhealthyRange and added `UnhealthyConditions` field

Reply:
It might make sense to have our own copy of the type instead of referencing the CAPI type